### PR TITLE
bugfix: fix issue#1369 prevent global session status inconsistency in cluster db mode

### LINKED ISSUE #1369: GlobalSession has concurrency problem in cluster mode?
## Why you need it?
Is it possible that more than one servers currently changing globalSession's status in cluster mode? If this situation happened, local ReentrantLock can not keep globalSession status changing procedure thread safe. Is it correct?
e.g., server A is commiting a transaction, meanwhile server B's timeout checking thread is trying to start a rollback. Is this a problem?

## How it could be?
Is optimistic locking on global_table a possible solution?

### PR DESCRIPTION
**Pull request by ggndnn**
### Ⅰ. Describe what this PR did
Prevent global session status inconsistency in cluster db mode.

### Ⅱ. Does this pull request fix one issue?

fixes #1369

### Ⅲ. Why don't you add test cases (unit test/integration test)? 
Did.

### CODE REVIEW

File: core/src/main/java/io/seata/core/store/db/LogStoreDataBaseDAO.java [hunk 1]
```diff
@@ -280,7 +281,12 @@ public boolean updateGlobalTransactionDO(GlobalTransactionDO globalTransactionDO
             ps = conn.prepareStatement(sql);
             ps.setInt(1, globalTransactionDO.getStatus());
             ps.setString(2, globalTransactionDO.getXid());
-            return ps.executeUpdate() > 0;
+            ps.setInt(3, globalTransactionDO.getVersion());
+            int cnt = ps.executeUpdate();
+            if(cnt != 1) {
+                throw new StoreException("GlobalSession was updated or deleted by another transaction");
```
Comment (CoffeeLatte007) [inline core/src/main/java/io/seata/core/store/db/LogStoreDataBaseDAO.java]: no retry?

Comment (ggndnn) [reply core/src/main/java/io/seata/core/store/db/LogStoreDataBaseDAO.java]: In my opinion, retry is not necessary.

Comment (ggndnn) [reply core/src/main/java/io/seata/core/store/db/LogStoreDataBaseDAO.java]: I think, ideally, throwing this exception is a small probability event. If failed, retry will triggered by high level code at a later time.

**Comment by ggndnn** [comment]
@CoffeeLatte007 Thanks for your review. Any more comments?

**Comment by funky-eyes** [comment]
Due to code conflicts and age, if you are still interested in participating in the community, please resubmit this pr to the 2.x branch after resolving the conflict.
由于代码冲突以及年代久远，如果你还有兴趣参与到社区，请解决完冲突后重新请提交这份pr至2.x分支中

**Comment by funky-eyes** [comment]
https://github.com/seata/seata/pull/5501

**Pull request closed without merging**